plugins/kubernetes/app/models/kubernetes/template_filler.rb

```diff
@@ -88,15 +88,18 @@ def namespaced_kind?(kind)
       api_version = template.fetch(:apiVersion)
 
       resources =
-        Rails.cache.fetch(["template-filler-resources", api_version, cluster]) do
+        Rails.cache.fetch(["template-filler-resources", api_version, cluster], expires_in: 1.hour) do
```

Comment:
This is slightly better, but it would be nice to reset cache if deploy contains new CR Definitions. 
Basically, check if there is a resource
```
apiVersion: apiextensions.k8s.io/v1
kind: CustomResourceDefinition
```